lib/model/model.go

```diff
@@ -2406,22 +2412,25 @@ func (m *model) RemoteSequence(folder string) (int64, bool) {
 	return ver, true
 }
 
-func (m *model) GlobalDirectoryTree(folder, prefix string, levels int, dirsonly bool) map[string]interface{} {
+func (m *model) GlobalDirectoryTree(folder, prefix string, levels int, dirsonly bool) []*DirectoryTree {
 	m.fmut.RLock()
 	files, ok := m.folderFiles[folder]
 	m.fmut.RUnlock()
 	if !ok {
 		return nil
 	}
 
-	output := make(map[string]interface{})
 	sep := string(filepath.Separator)
 	prefix = osutil.NativeFilename(prefix)
 
 	if prefix != "" && !strings.HasSuffix(prefix, sep) {
 		prefix = prefix + sep
 	}
 
+	output := DirectoryTree{Children: []*DirectoryTree{}}
+	breadcrumbs := []*DirectoryTree{&output}
+
+	var currentPrefix []string
 	files.WithPrefixedGlobalTruncated(prefix, func(fi db.FileIntf) bool {
```

Comment:
This also does not return what you expect.

This returns all files advertised by all devices, some which we might not have, or some which we might ignored by us.

The original purpose of this function was to be able to make a browsable tree for users to see all files available across all devices, select which folders they want to sync by selecting them in the tree, and based on that adjusting the ignores file.


So I guess the question is, what exactly are we trying to answer with this feature. All files available, local files available? Local not ignored files? All not ignored files?

Reply:
Well yes, I didn't expect that 😀I just returned a structure which came in a way of paths in strings. (Didn't include files if `dirsOnly` was set to True) I programmed it to make the existing tests pass. I thought to make the tests pass while changing the resulting structure would be enough.
I thought that this resulting structure should contain local, not ignored files (I forgot to not include the ignored ones) from one shared device's directory (for example `default` or other, not all of them). 
The piece of code where we sent all of the files available across all devices to `WithPrefixedGlobalTruncated` is already done? (I didn't detect such piece of code.) Or should It be done as well?

Reply:
WithPrefixedGlobal is the function that gives you all the files available from all devices. It does not take local ignoring into account however.

Reply:
I think you can keep your changes and just create two wrapper functions `GlobalDirectoryTree` and `LocalDirectoryTree`, which under the hood both do the same thing but use `files.WithPrefixedGlobalTruncated` or `files.WithPrefixedHaveTruncated` (with local device ID) respectively. That could even be extended to a `RemoteDirectoryTree(...remoteID...)` which returns the files as seen by the remote devices (probably unnecessary though).

Reply:
Actually checking which files which device has might be a nice feature.